consul/data_source_consul_agent_config.go

```diff
@@ -51,7 +51,10 @@ func dataSourceConsulAgentConfig() *schema.Resource {
 }
 
 func dataSourceConsulAgentConfigRead(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*consulapi.Client)
+	client, err := meta.(*Config).Client()
+	if err != nil {
+		return err
```

Comment:
Should we prefix these errors? I wonder if it will be obvious what caused them.

Reply:
I put those error checks to cover all possibilities but this should never happen since the config is already checked during the provider configuration at https://github.com/terraform-providers/terraform-provider-consul/pull/105/files#diff-b338f46ba92504078fcef3e0e3a1b2cbR127. I could add a helper function to make this more obvious:

```
func getClient(meta interface{}) consulapi.Client {
    // ignore error as the configuration has already been checked in providerConfigure()
    client, _ := meta.(*Config).Client()
    return client
}
```
and replace all the 
```
	client, err := meta.(*Config).Client()
	if err != nil {
		return err
	}
```
by 
```
     client := getClient(meta)
```